BaikalDataGenerator/Source/render.cpp

```diff
@@ -138,179 +121,90 @@ Render::Render(const std::filesystem::path& scene_file,
     }
 
     m_renderer->SetMaxBounces(m_num_bounces);
-
-    if (!std::filesystem::exists(scene_file))
-    {
-        THROW_EX("There is no any scene file to load");
-    }
-
-    // workaround to avoid issues with tiny_object_loader
-    auto scene_dir = scene_file.parent_path().string();
-
-    if (scene_dir.back() != '/' || scene_dir.back() != '\\')
-    {
-#ifdef WIN32
-        scene_dir.append("\\");
-#else
-        scene_dir.append("/");
-#endif
-    }
-
-    m_scene = SceneIo::LoadScene(scene_file.string(), scene_dir);
-
-    // load materials.xml if it exists
-    auto materials_file = scene_file.parent_path() / "materials.xml";
-    auto mapping_file = scene_file.parent_path() / "mapping.xml";
-
-    if (std::filesystem::exists(materials_file) &&
-        std::filesystem::exists(mapping_file))
-    {
-        auto material_io = MaterialIo::CreateMaterialIoXML();
-        auto materials = material_io->LoadMaterials(materials_file.string());
-        auto mapping = material_io->LoadMaterialMapping(mapping_file.string());
-
-        material_io->ReplaceSceneMaterials(*m_scene, *materials, mapping);
-    }
-    else
-    {
-        std::cout << "WARNING: materials.xml or mapping.xml is missed" << std::endl;
-    }
 }
 
-
-void Render::SaveMetadata(const std::filesystem::path& output_dir) const
+void Render::AttachLight(LightObject* light)
 {
-    using namespace tinyxml2;
+    m_scene->AttachLight(light);
+}
 
-    XMLDocument doc;
+void Render::SaveMetadata(const std::filesystem::path& output_dir,
+                          const std::string& scene_name,
+                          unsigned cameras_start_idx,
+                          unsigned cameras_end_idx,
+                          int cameras_index_offset,
+                          bool gamma_correction_enabled) const
+{
+    tinyxml2::XMLDocument doc;
 
     auto file_name = output_dir;
     file_name.append("metadata.xml");
 
-    XMLNode* root= doc.NewElement("metadata");
+    auto* root = doc.NewElement("metadata");
     doc.InsertFirstChild(root);
 
-    XMLElement* scene = doc.NewElement("scene");
-    scene->SetAttribute("file", m_scene_file.c_str());
+    auto* scene = doc.NewElement("scene");
+    scene->SetAttribute("name", scene_name.c_str());
     root->InsertEndChild(scene);
 
-    // log outputs layout
-    XMLElement* size_attribute = doc.NewElement("layout");
-    size_attribute->SetAttribute("width", m_width);
-    size_attribute->SetAttribute("height", m_height);
-    root->InsertEndChild(size_attribute);
+    auto* cameras = doc.NewElement("cameras");
+    cameras->SetAttribute("start_idx", static_cast<int>(cameras_start_idx));
+    cameras->SetAttribute("end_idx", static_cast<int>(cameras_end_idx));
+    cameras->SetAttribute("idx_offset", cameras_index_offset);
+    root->InsertEndChild(cameras);
```

Comment:
After removing parameter cameras_start_idx we don't need node "cameras" at all. But we have to put cameras info into dedicated xml file (i.e. app_metadata.xml) to be able handle it on the server side.